stdpopsim/slim_engine.py

```diff
@@ -201,6 +204,52 @@
      * Register events occurring at time T_start or more recently.
      */
 
+    // Save/restore events. These should come before all other events.
+    if (length(drawn_mutations) > 0) {
+        for (i in 0:(ncol(drawn_mutations)-1)) {
+            save = drawn_mutations[4,i] == 1;
+            if (!save) {
+                next;
+            }
+            g = G_start + gdiff(T_start, drawn_mutations[0,i]);
+            // Unconditionally save the state before the mutation is drawn.
+            sim.registerLateEvent(NULL, "{save();}", g, g);
```

Comment:
what is this `save()` doing? I can't find it anywhere in the SLiM/Eidos manuals.